sdk/storage/azure-storage-blobs/test/ut/blob_batch_client_test.cpp

```diff
@@ -0,0 +1,254 @@
+      sasBuilder.ExpiresOn = std::chrono::system_clock::now() + std::chrono::minutes(5);
+      sasBuilder.BlobContainerName = containerName;
+      sasBuilder.Resource = Sas::BlobSasResource::BlobContainer;
+      sasBuilder.SetPermissions(Sas::BlobContainerSasPermissions::All);
+      return sasBuilder.GenerateSasToken(
+          *_internal::ParseConnectionString(StandardStorageConnectionString()).KeyCredential);
+    }();
+
+    auto serviceClient = GetClientForTest(testName);
+    serviceClient.GetBlobContainerClient(containerName).CreateIfNotExists();
+    auto containerClient = Blobs::BlobContainerClient(
+        serviceClient.GetBlobContainerClient(containerName).GetUrl() + containerSasToken,
+        InitClientOptions<Blobs::BlobClientOptions>());
+    auto blob1Client = containerClient.GetBlockBlobClient(blob1Name);
+    blob1Client.UploadFrom(nullptr, 0);
+    auto blob2Client = containerClient.GetBlockBlobClient(blob2Name);
+    blob2Client.UploadFrom(nullptr, 0);
+
+    auto batch = containerClient.CreateBatch();
+    auto setTier1Response
+        = batch.SetBlobAccessTier(containerName, blob1Name, Blobs::Models::AccessTier::Cool);
+    auto setTier2Response
+        = batch.SetBlobAccessTier(blob2Client.GetUrl(), Blobs::Models::AccessTier::Archive);
+    auto submitBatchResponse = containerClient.SubmitBatch(batch);
+
+    EXPECT_NO_THROW(setTier1Response.GetResponse());
+    EXPECT_NO_THROW(setTier2Response.GetResponse());
+    EXPECT_EQ(
+        blob1Client.GetProperties().Value.AccessTier.Value(), Blobs::Models::AccessTier::Cool);
+    EXPECT_EQ(
+        blob2Client.GetProperties().Value.AccessTier.Value(), Blobs::Models::AccessTier::Archive);
+
+    serviceClient.DeleteBlobContainer(containerName);
+  }
+
+  TEST_F(BlobBatchClientTest, TokenAuthorization)
+  {
+    const std::string testName = GetTestNameLowerCase();
+
+    std::shared_ptr<Azure::Core::Credentials::TokenCredential> credential
+        = std::make_shared<Azure::Identity::ClientSecretCredential>(
+            AadTenantId(), AadClientId(), AadClientSecret());
+    Blobs::BlobClientOptions options;
+
+    auto serviceClient = InitTestClient<Blobs::BlobServiceClient, Blobs::BlobClientOptions>(
+        GetClientForTest(testName).GetUrl(), credential, options);
+
+    const std::string containerName = testName;
+    const std::string blobName = "b1";
+
+    auto containerClient = serviceClient->GetBlobContainerClient(containerName);
+    containerClient.CreateIfNotExists();
+    auto blobClient = containerClient.GetAppendBlobClient(blobName);
+    blobClient.Create();
+
+    auto batch = containerClient.CreateBatch();
+    auto delete1Response = batch.DeleteBlob(blobClient.GetUrl());
+    auto submitBatchResponse = containerClient.SubmitBatch(batch);
+
+    EXPECT_TRUE(delete1Response.GetResponse().Value.Deleted);
+
+    containerClient.Delete();
+  }
+
+  TEST_F(BlobBatchClientTest, Exceptions_LIVEONLY_)
```

Comment:
Any test done for mixed result case? e.g. subrequest 1/2 success, subrequest 3 failed.